test/e2e/e2e_test.go

```diff
@@ -177,6 +178,145 @@ func TestLowNodeUtilization(t *testing.T) {
+	if err != nil {
+		t.Errorf("Error during client creation with %v", err)
+	}
+
+	stopChannel := make(chan struct{}, 0)
+
+	sharedInformerFactory := informers.NewSharedInformerFactory(clientSet, 0)
+	sharedInformerFactory.Start(stopChannel)
+	sharedInformerFactory.WaitForCacheSync(stopChannel)
+	defer close(stopChannel)
+
+	nodeInformer := sharedInformerFactory.Core().V1().Nodes()
+
+	testNamespace := &v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "e2e-" + strings.ToLower(t.Name())}}
+	if _, err := clientSet.CoreV1().Namespaces().Create(ctx, testNamespace, metav1.CreateOptions{}); err != nil {
+		t.Fatalf("Unable to create ns %v", testNamespace.Name)
+	}
+	defer clientSet.CoreV1().Namespaces().Delete(ctx, testNamespace.Name, metav1.DeleteOptions{})
+
+	rc := RcByNameContainer("test-rc-toomanyrestarts", testNamespace.Name, 5, map[string]string{"test": "toomanyrestarts"}, nil)
+	if _, err := clientSet.CoreV1().ReplicationControllers(rc.Namespace).Create(ctx, rc, metav1.CreateOptions{}); err != nil {
+		t.Errorf("Error creating deployment %v", err)
+	}
+	defer deleteRC(ctx, t, clientSet, rc)
+
+	// wait for a while so all the pods are at least few seconds older
+	time.Sleep(5 * time.Second)
+
+	// it's assumed all new pods are named differently from currently running -> no name collision
+	podList, err := clientSet.CoreV1().Pods(rc.Namespace).List(ctx, metav1.ListOptions{LabelSelector: labels.SelectorFromSet(rc.Spec.Template.Labels).String()})
+	if err != nil {
+		t.Fatalf("Unable to list pods")
+	}
+
+	if len(podList.Items) != 5 {
+		t.Fatalf("Expected 5 replicas, got %v instead", len(podList.Items))
+	}
+
+	podNamesInclude := make(map[string]struct{})
+	for _, pod := range podList.Items {
+		podNamesInclude[pod.Name] = struct{}{}
+	}
+
+	t.Logf("Existing pods: %v", getPodNames(podList.Items))
+	t.Logf("set the strategy to delete pods from %v namespace", rc.Namespace)
+	runPodLifetimeStrategy(ctx, clientSet, nodeInformer, deschedulerapi.Namespaces{
+		Include: []string{rc.Namespace},
+	})
+
+	t.Logf("Waiting 10s")
+	time.Sleep(10 * time.Second)
+	podList, err = clientSet.CoreV1().Pods(rc.Namespace).List(ctx, metav1.ListOptions{LabelSelector: labels.SelectorFromSet(rc.Spec.Template.Labels).String()})
+	if err != nil {
+		t.Fatalf("Unable to list pods after running strategy")
+	}
+
+	t.Logf("Existing pods: %v", getPodNames(podList.Items))
+
+	podNamesExclude := make(map[string]struct{})
+	// validate all pod were deleted
+	for _, pod := range podList.Items {
+		podNamesExclude[pod.Name] = struct{}{}
+		if _, exists := podNamesInclude[pod.Name]; exists {
+			t.Fatalf("Pod %q is expected to be deleted", pod.Name)
+		}
+	}
+
+	t.Logf("set the strategy to delete pods from namespaces except the %v namespace", rc.Namespace)
+	runPodLifetimeStrategy(ctx, clientSet, nodeInformer, deschedulerapi.Namespaces{
+		Exclude: []string{rc.Namespace},
+	})
+
+	t.Logf("Waiting 10s")
+	time.Sleep(10 * time.Second)
```

Comment:
another duplicate `Sleep`, though I'm not sure why you need to wait after running the strategy? Just to make sure the pods are deleted?

Reply:
Here, I am waiting for some time (assuming the strategy finished running in that time and all pods that were supposed to be deleted got deleted). 10 seconds is just a guess. Though, at this point it's impossible to decide if a strategy run properly and did not deleted any pod. Or, if the strategy failed running for some reason but did not report any error.

Reply:
Ah, this is the exclude test. So you're testing that none of these pods get deleted. I guess there isn't really a condition we can Poll for then